src/nbl/CMakeLists.txt

```diff
@@ -576,132 +576,129 @@ if(_NBL_COMPILE_WITH_GLI_)
 endif()
 
 # DXC
-if(_NBL_COMPILE_WITH_DXC_)
-	add_dependencies(Nabla dxc)
-	target_include_directories(Nabla PRIVATE 
-		"${DXC_BINARY_DIR_GE}/tools/clang/tools/dxcompiler"
-		"${DXC_SOURCE_DIR}/tools/clang/tools/dxcompiler"
-		"${DXC_SOURCE_DIR}/tools/clang/include"
-		"${DXC_BINARY_DIR_GE}/tools/clang/include"
-		"${DXC_SOURCE_DIR}/include/llvm/llvm_assert"
-		"${DXC_BINARY_DIR_GE}/include"
-		"${DXC_SOURCE_DIR}/include"
-		"${DXC_BINARY_DIR_GE}/include/dxc/Tracing"
-		"${DXC_BINARY_DIR_GE}/utils/version"
-	)
-	target_link_libraries(Nabla INTERFACE "${DXC_DXCOMPILER_LIB}")
+add_dependencies(Nabla dxc)
+target_include_directories(Nabla PRIVATE 
+	"${DXC_BINARY_DIR_GE}/tools/clang/tools/dxcompiler"
+	"${DXC_SOURCE_DIR}/tools/clang/tools/dxcompiler"
+	"${DXC_SOURCE_DIR}/tools/clang/include"
+	"${DXC_BINARY_DIR_GE}/tools/clang/include"
+	"${DXC_SOURCE_DIR}/include/llvm/llvm_assert"
+	"${DXC_BINARY_DIR_GE}/include"
+	"${DXC_SOURCE_DIR}/include"
+	"${DXC_BINARY_DIR_GE}/include/dxc/Tracing"
+	"${DXC_BINARY_DIR_GE}/utils/version"
+)
+target_link_libraries(Nabla INTERFACE "${DXC_DXCOMPILER_LIB}")
+
+############################################################################
+# list files you want to compile with DXC in _NBL_ENABLE_DXC_COMPILE_TESTS_
+# once we have guarantee all of shaders in Nabla can be compiled with DXC,
+# we can simply grep all of those files/use existing vars holding paths to
+# shaders and remove manual adding the files bellow
+###################################################
+
+if(_NBL_ENABLE_DXC_COMPILE_TESTS_)
+	set(_NBL_DXC_CT_INCLUDE_ROOT_ "${NBL_ROOT_PATH}/include") # Nabla include root path for shaders
+	set(_NBL_DXC_CT_BINARY_BIN_ "${CMAKE_CURRENT_BINARY_DIR}/DXC/CT") # compile-test directory
```

Comment:
can we move this testing crap to a separate `CMakeLists.txt` ?

Reply:
resolved